terraform/snowflake/environments/dev/main.tf

```
@@ -104,3 +109,25 @@ resource "snowflake_grant_account_role" "logger_to_transformer" {
   role_name        = "LOGGER_${var.environment}"
   parent_role_name = "TRANSFORMER_${var.environment}"
 }
+
+# Default user password policy
+resource "snowflake_password_policy" "user_password_policy" {
```

This probably defaults to the "PUBLIC" provider above, which won't have permissions to set this policy.
It might need something like the following:
```suggestion
resource "snowflake_password_policy" "user_password_policy" {
  provider = snowflake.securityadmin
```

---

Thank you Ian ! Good catch !